services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/utils/SubnetManagementUtil.java

```diff
@@ -66,4 +76,134 @@ public static boolean checkSubnetRequestResourceIsValid(SubnetRequestWebJson res
         return true;
     }
 
+    public static String setGatewayIpValue(String gatewayIp, String cidr) {
+        // gatewayIP is null
+        if (gatewayIp == null) {
+            return null;
+        }
+
+        // gatewayIP is empty
+        if (gatewayIp.length() == 0) {
+            SubnetUtils utils = new SubnetUtils(cidr);
+            String lowIp = utils.getInfo().getLowAddress();
+            if (lowIp == null) {
+                return null;
+            }
+
+            String[] lowIps = lowIp.split("\\.");
+            Integer low = Integer.parseInt(lowIps[lowIps.length - 1]) + ConstantsConfig.BaseInterval;
+            lowIps[lowIps.length - 1] = String.valueOf(low);
+            lowIp = String.join(".", lowIps);
+            return lowIp;
+        }
+
+        return gatewayIp;
+
+    }
+
+    public static boolean checkGatewayIpIsValid(String gatewayIp, String cidr) throws ParameterUnexpectedValueException, FallbackException {
```

Comment:
I would recommend to rename it to "checkGatewayIpInputSupported". This is because if an input is NULL or empty, although they are not "valid" IPs, but they are valid input value for the GW IP parameter.

Reply:
Sure